server/grpc/gogoreflection/fix_registration.go

```diff
@@ -0,0 +1,143 @@
+package gogoreflection
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"reflect"
+
+	_ "github.com/gogo/protobuf/gogoproto" // required so it does register the gogoproto file descriptor
+	gogoproto "github.com/gogo/protobuf/proto"
+
+	// nolint: staticcheck
+	"github.com/golang/protobuf/proto"
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	_ "github.com/regen-network/cosmos-proto" // look above
+)
+
+var importsToFix = map[string]string{
+	"gogo.proto":   "gogoproto/gogo.proto",
+	"cosmos.proto": "cosmos_proto/cosmos.proto",
+}
+
+// fixRegistration is required because certain files register themselves in a way
+// but are imported by other files in a different way.
+// NOTE(fdymylja): This fix should not be needed and should be addressed in some CI.
+// Currently every cosmos-sdk proto file is importing gogo.proto as gogoproto/gogo.proto,
+// but gogo.proto registers itself as gogo.proto, same goes for cosmos.proto.
```

Comment:
Do you know why this happens? Maybe we should fix it in `github.com/regen-network/cosmos-proto`? 
@aaronc

Reply:
I mean - let's create an issue if there is a bug we can easily fix in cosmos-proto

Reply:
it's not a bug per se, it's a shortcoming when it comes to combo together reflection and proto imports. protofiles register themselve in the registry with a path, but then are imported with another path. this makes reflection fail, because it cannot find certain files.

this would mean either forking gogoproto and changing how we feed the file to protoc, or changing the third_party structure in the sdk, edit how every file imports gogoproto. 

and im not sure if this is a breaking change, descriptor wise it is.

probably this should be a blob of work which should be reorganized during the transition to protov2